### PR TITLE
Allow multiple schedulers in helm chart

### PR DESCRIPTION
**Pull request by dimberman**
(cherry picked from commit 7f06d6220747151f527ab372b6ce431c61f6f20f)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -288,6 +288,7 @@ workers:
 
 # Airflow scheduler settings
 scheduler:
+  replicas: 2
```
Comment (ashb) [inline chart/values.yaml]:
```suggestion
  replicas: 1
```

Default needs to stay at 1 I think

File: chart/values.yaml [hunk 2]
```diff
@@ -28,10 +28,10 @@ gid: 50000
 airflowHome: "/opt/airflow"
 
 # Default airflow repository -- overrides all the specific images below
-defaultAirflowRepository: apache/airflow
+defaultAirflowRepository: dimberman/airflow
 
 # Default airflow tag to deploy
-defaultAirflowTag: 1.10.12
+defaultAirflowTag: scheduler-ha-with-dags-0.1
```
Comment (ashb) [inline chart/values.yaml]: Need to revert these :)

File: chart/templates/scheduler/scheduler-deployment.yaml [hunk 3]
```diff
@@ -46,7 +46,7 @@ spec:
 {{- if $stateful }}
   serviceName: {{ .Release.Name }}-scheduler
 {{- end }}
-  replicas: 1
+  replicas: {{ .Values.scheduler.replicas }}
```
Comment (ashb) [inline chart/templates/scheduler/scheduler-deployment.yaml:49]: I wonder if there's any way we can guard this to only work for 2.0.0 -- but given users might use their own Docker image I don't think there's much we can really do.

I guess just make it clear in the values.yaml file do not set this >1 unless you are on 2.0.0.

**Review by ashb — CHANGES_REQUESTED** [review]

File: chart/templates/scheduler/scheduler-deployment.yaml (hunk 3 above)
Comment (dimberman) [reply chart/templates/scheduler/scheduler-deployment.yaml:49]: @ashb yeah I don't think there's a warning, maybe we just need to be clear in documentation :/.

**Review by ashb — DISMISSED** [review]
Need to add these new values to the schema too please

File: chart/values.yaml [hunk 4]
```diff
@@ -289,6 +289,7 @@ workers:
 
 # Airflow scheduler settings
 scheduler:
+  replicas: 1
```
Comment (ashb) [inline chart/values.yaml:294]: Could you add a comment here about Mysql5 or Airflow 1.10

Comment (dimberman) [reply chart/values.yaml:294]: Sure thing

Comment (dimberman) [reply chart/values.yaml:294]: @ashb fixed

**Review by kaxil — COMMENTED** [review]
Can you please update https://github.com/apache/airflow/blob/master/chart/README.md#parameters too

**Comment by dimberman** [comment]
@kaxil updated

**Review by kaxil — APPROVED** [review]

**Comment by mik-laj** [comment]
This change seems to me not enough to support HA. We should also add an improvement to the liveness probe because it currently only watches the newest job, not the job assigned to the current scheduler.
https://github.com/apache/airflow/blob/91a64db505e50712cd53928b4f2b84aece3cc1c0/chart/templates/scheduler/scheduler-deployment.yaml#L115-L130

**Pull request merged**